nixos/modules/services/networking/harmonia.nix

```diff
@@ -0,0 +1,95 @@
+{ config, pkgs, lib, ... }:
+let
+  cfg = config.services.harmonia;
+
+  format = pkgs.formats.toml { };
+  configFile = format.generate "harmonia.toml" cfg.settings;
+in
+{
+  options = {
+    services.harmonia = {
+      enable = lib.mkEnableOption (lib.mdDoc "Harmonia: Nix binary cache written in Rust");
+
+      signKeyPath = lib.mkOption {
+        type = lib.types.nullOr lib.types.path;
+        default = null;
+        description = lib.mdDoc "Path to the signing key to use for signing the cache";
+      };
+
+      package = lib.mkPackageOptionMD pkgs "harmonia" { };
+
+      settings = lib.mkOption {
+        inherit (format) type;
+
+        description = lib.mdDoc "Settings to merge with the default configuration";
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    systemd.services.harmonia = {
+      description = "harmonia binary cache service";
+
+      requires = [ "nix-daemon.socket" ];
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+
+      environment = {
+        NIX_REMOTE = "daemon";
+        LIBEV_FLAGS = "4"; # go ahead and mandate epoll(2)
+        CONFIG_FILE = lib.mkIf (configFile != null) configFile;
+        SIGN_KEY_PATH = lib.mkIf (cfg.signKeyPath != null) "%d/sign-key";
+        RUST_LOG = "info";
```

Comment:
Maybe make a logging option? That could be helpful for debugging.

Reply:
Make an option called logLevel, there are many other modules in nixpkgs that used RUST_LOG

Reply:
We can do this if someone actually needs this for harmonia. I don't see a use case for this particular project (there are not many debug statements in the code)

For now one can also do:

```
systemd.services.harmonia.environment.RUST_LOG = "debug";
```

Reply:
Alright maybe set it as a mkDefault so people don't need to mkForce